test/plugins/posthtml-flatten-script.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+var assert = require('power-assert');
+var Bluebird = require('bluebird');
+var posthtml = require('posthtml');
+var describe = require('mocha').describe;
+var it = require('mocha').it;
+
+var plugin = require('../../lib/plugins/posthtml-flatten-script');
+
+function test(input, output, opts) {
+  return posthtml([plugin(opts)]).process(input).then(function (result) {
+    assert(result.html === output);
+  });
+}
+
+describe('posthtml-flatten-script', function () {
+  it('should flatten inline JavaScript', function () {
+    return test('<script>alert("foo" + "bar");</script>', '<script>alert("foobar");</script>', {
```

Comment:
Please add a test for `"</scr" + "ipt>"`. If they're concatenated after minification, it will break the script tag.